tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs

```diff
@@ -0,0 +1,128 @@
+﻿namespace FSharpVSPowerTools.Tests
+
+open System
+open NUnit.Framework
+open FSharpVSPowerTools
+open FSharpVSPowerTools.Navigation
+open FSharpVSPowerTools.ProjectSystem
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio
+open Microsoft.VisualStudio.OLE.Interop
+open Microsoft.VisualStudio.Shell.Interop
+
+type GoToDefinitionCommandHelper() =    
+    inherit VsTestBase()
+    let command = new GoToDefinitionFilterProvider(
+                        editorFactory = base.VsEditorAdaptersFactoryService,
+                        textDocumentFactoryService = base.DocumentFactoryService,                            
+                        fsharpVsLanguageService = base.VsLanguageService,
+                        serviceProvider = base.ServiceProvider,
+                        projectFactory = base.ProjectFactory,
+                        referenceSourceProvider = base.ReferenceSourceProvider)
+
+    member __.GetCommandFilter(wpfTextView) =
+        command.RegisterCommandFilter(wpfTextView, fireNavigationEvent = true)
+
+module GoToDefinitionCommandTests =
+    open System.IO
+
+#if APPVEYOR
+    let timeout = 60000<ms>
+#else
+    let timeout = 10000<ms>
+#endif
+
+    let helper = GoToDefinitionCommandHelper()
+   
+    [<TestFixtureSetUp>]
+    let setUp() =
+        TestUtilities.AssertListener.Initialize()
+        DocumentEventListener.SkipTimerDelay <- true
+        Logger.GlobalServiceProvider <- helper.ServiceProvider
+
+    [<Test>]
+    let ``should be able to go to definition to an external class``() = 
+        let content = """
+module NavigateToSourceTests
+open System.IO
+let f x = Path.GetFileName(x)
+let g x = File.Exists(x)
+"""
+        let projectFileName = fullPathBasedOnSourceDir "../data/NavigateToSource/NavigateToSource.fsproj"
+        let fileName = fullPathBasedOnSourceDir "../data/NavigateToSource/OctokitTests.fs"
+        let buffer = createMockTextBuffer content fileName
+        helper.SetUpProjectAndCurrentDocument(ExternalProjectProvider(projectFileName), fileName)              
+        let textView = createMockTextView buffer
+        let command = helper.GetCommandFilter(textView)
+        let urlChanged = command.UrlChanged.Value
+        let filter = command :> IOleCommandTarget
+        let prefix = Path.GetFullPath(fileName)
```

Comment:
These 9 lines are duplicated in all three tests. Maybe it's a good idea to extract them into a `SetUp` or a plane function?